offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/offergeneration/QuotationDetails.groovy

```diff
@@ -0,0 +1,449 @@
+        String elementId = generateElementID(0, productGroup)
+        htmlContent.getElementById(elementId).empty()
+
+        if (isOverflowingPage()) {
+            generateHTMLTableOnNextPage("${productGroup.toString()}-table")
+            resetPageItemsCount()
+        }
+        //Append Table Title and Header
+        htmlContent.getElementById(elementId).append(ItemPrintout.tableHeader())
+
+        items.each { ProductItem item ->
+            generateItemContent(item, elementId)
+        }
+        //account for spaces of added table elements, footer, totals,...
+        pageItemsCount += 4
+        htmlContent.getElementById(elementId).append(ItemPrintout.subTableFooter(productGroup))
+
+        // Update footer Prices
+        addSubTotalPrices(productGroup, items)
+    }
+
+    /**
+     * Appends an item to a element id, if no space is left on the page a new page is and table start is created
+     * @param item The item place into the HTML document
+     * @param elementId The id references where the item is added
+     */
+    private void generateItemContent(ProductItem item, String elementId){
+        itemNumber++
+        if (isOverflowingPage()) {
+            generateHTMLTableOnNextPage(elementId)
+            //repeat table header for next page
+            htmlContent.getElementById(elementId).append(ItemPrintout.tableHeader())
+            resetPageItemsCount()
+        }
+
+        //add product to current table
+        htmlContent.getElementById(elementId).append(ItemPrintout.itemInHTML(itemNumber, item))
+        pageItemsCount += determineItemSpace(item)
+    }
+
+    private static String generateElementID(int tableCount, ProductGroup productGroups){
+        return productGroups.toString() + "-" + "product-items" + "-" + ++tableCount}
+
+    private void resetPageItemsCount(){ pageItemsCount = 1}
+
+    /**
+     * Creates a new table element after a page break
+     * @param elementId The id which will reference the new table
+     */
+    private void generateHTMLTableOnNextPage(String elementId){
+        htmlContent.getElementById(elementId).append(ItemPrintout.pageBreak())
+        elementId = "product-items" + "-" + ++tableCount
+        htmlContent.getElementById("item-table-grid").append(ItemPrintout.createNewTable(elementId))
+    }
+
+    private boolean isOverflowingPage() {
+        return pageItemsCount >= maxPageItems
+    }
+
+    /**
+     * Adds the subtotal NET price for a product group to the HTML template
+     * @param productGroup The product group for which the subtotal is added
+     * @param productItems The product items for which the NET is calculated
+     */
+    private void addSubTotalPrices(ProductGroup productGroup, List<ProductItem> productItems) {
+        double netSum = calculateNetSum(productItems)
+        final netPrice = Currency.getFormatterWithoutSymbol().format(netSum)
+
+        htmlContent.getElementById("${productGroup}-net-costs-value").text(netPrice)
+    }
+
```

Comment:
Missing javadoc 
```suggestion
    /**
     * Helper method that calculates the NET price for a list of productItems      
     * @param productItems The product item list for which the NET is calculated
     */
```

Reply:
We would also need to specify the return value here (NIT) could you extend your suggestion @Steffengreiner

Reply:
```suggestion
          /**
	     * Helper method that calculates the NET price for a list of product items      
	     * @param productItems The product item list for which the NET is calculated
	     *  @return The net value for the given list of items 
	     */
```